Related to Issue #373, allowing absolute references for self (#414)
radiantearth/stac-spec: item-spec/item-spec.md

with the links section, to describe things like the catalog an item is in, related items, parent or
child items (modeled in different ways, like an 'acquisition' or derived data).
It is allowed to add additional fields such as a `title` and `type`.

| Field Name | Type   | Description                                                                                                                         |
| ---------- | ------ | ----------------------------------------------------------------------------------------------------------------------------------- |
| href       | string | **REQUIRED.** The actual link in the format of an URL. Relative and absolute links are both allowed unless otherwise noted in the ```rel``` type description.                    |
| rel        | string | **REQUIRED.** Relationship between the current document and the linked document. See chapter "Relation types" for more information. |
| type       | string | Media type of the referenced entity. 																								|
| title      | string | A human readable title to be used in rendered displays of the link. |

#### Relative vs Absolute links

> Currently the JSON schema for links does not require a URI formatting, to give the option for

Reviewer: ```suggestion
Currently, the JSON schema for links does not require them to be formatted as URIs, to allow
```

Reviewer 2: It sounds like we should be able to use another JSON construct (uri-ref or something? @m-mohr was talking about this) so the schema can validate relative and absolute links.

> implementors to provide relative links. In general, Catalog API's should aim for absolute links

Reviewer: ```suggestion
implementors to provide relative links. In general, Catalog APIs should aim to provide absolute links
```

> whenever possible. But Static Catalogs are potentially more portable if they can be implemented with

Reviewer: ```suggestion
whenever possible. Static Catalogs are potentially more portable if they incorporate only
```

relative links, so that every link doesn't need to be rewritten when the data is copied. Additional
recommendations for particular ```rel``` types are given in the ```rel``` type description.

#### Relation types

The following types are commonly used as `rel` types in the Link Object of an Item:

| Type         | Description                                                  |
| ------------ | ------------------------------------------------------------ |

> | self         | STRONGLY RECOMMENDED. _Absolute_ URL to the Item if it is publicly online. This is particularly useful when in a download package that includes metadata, so that the downstream user can know where the data has come from. |

Reviewer: ```suggestion
| self         | STRONGLY RECOMMENDED. _Absolute_ URL to the Item if it is available at a public URL. This is particularly useful when in a download package that includes metadata, so that the downstream user can know where the data has come from. |
```

| Type         | Description                                                  |
| ------------ | ------------------------------------------------------------ |

> | self         | **REQUIRED.** _Absolute_ URL to the Item if it is publicly online, or an _absolute reference_ to the item in the form ```/path/to/item``` otherwise. This is particularly useful when in a download package that includes metadata, so that the downstream user can know where the data has come from. |

Reviewer 3: Shouldn't that be called "absolute path" instead of `absolute reference`?
